h5pt.go

```diff
@@ -82,170 +97,143 @@ func (t *Table) ReadPackets(start, nrecords int, data interface{}) error {
 	return h5err(err)
 }
 
-func extractStructValues(rv reflect.Value, rt reflect.Type) (ptr unsafe.Pointer, err error) {
-	// Initially the memory will be allocated to 64 bytes. If more is required
-	// later, this will be increased.
-	memSize := C.size_t(64)
-	ptr = C.malloc(memSize)
-
-	for i := 0; i < rv.NumField(); i++ {
-		var dataPtr unsafe.Pointer
-		var dataSize C.size_t
-		f := rv.Field(i)
-		ft := f.Type().Kind()
-		offset := C.size_t(rt.Field(i).Offset)
-
-		switch ft {
-		case reflect.Int8:
-			val := C.int8_t(int8(f.Int()))
-			dataPtr = unsafe.Pointer(&val)
-			dataSize = C.size_t(unsafe.Sizeof(val))
-		case reflect.Uint8:
-			val := C.uint8_t(uint8(f.Uint()))
-			dataPtr = unsafe.Pointer(&val)
-			dataSize = C.size_t(unsafe.Sizeof(val))
-		case reflect.Int16:
-			val := C.uint8_t(int16(f.Int()))
-			dataPtr = unsafe.Pointer(&val)
-			dataSize = C.size_t(unsafe.Sizeof(val))
-		case reflect.Uint16:
-			val := C.uint8_t(uint16(f.Uint()))
-			dataPtr = unsafe.Pointer(&val)
-			dataSize = C.size_t(unsafe.Sizeof(val))
-		case reflect.Int32:
-			val := C.uint8_t(int32(f.Int()))
-			dataPtr = unsafe.Pointer(&val)
-			dataSize = C.size_t(unsafe.Sizeof(val))
-		case reflect.Uint32:
-			val := C.uint8_t(uint32(f.Uint()))
-			dataPtr = unsafe.Pointer(&val)
-			dataSize = C.size_t(unsafe.Sizeof(val))
-		case reflect.Int64:
-			val := C.int64_t(int64(f.Int()))
-			dataPtr = unsafe.Pointer(&val)
-			dataSize = C.size_t(unsafe.Sizeof(val))
-		case reflect.Uint64:
-			val := C.uint64_t(uint64(f.Uint()))
-			dataPtr = unsafe.Pointer(&val)
-			dataSize = C.size_t(unsafe.Sizeof(val))
-		case reflect.Float32:
-			val := C.float(float32(f.Float()))
-			dataPtr = unsafe.Pointer(&val)
-			dataSize = C.size_t(unsafe.Sizeof(val))
-		case reflect.Float64:
-			val := C.double(float64(f.Float()))
-			dataPtr = unsafe.Pointer(&val)
-			dataSize = C.size_t(unsafe.Sizeof(val))
-		case reflect.Bool:
-			val := C.uchar(0)
-			if f.Bool() {
-				val = 1
-			}
-			dataPtr = unsafe.Pointer(&val)
-			dataSize = C.size_t(unsafe.Sizeof(val))
-		default:
-			err = fmt.Errorf("hdf5: Could not append struct member %s "+
-				"to PacketTable", ft)
-			return nil, err
-		}
-
-		// If the earlier allocated memory is not enough, we will increase it
-		// by another 64 bytes.
-		if memSize < (dataSize + offset) {
-			memSize += 64
-			ptr = C.realloc(ptr, memSize)
-			C.memset(unsafe.Pointer(uintptr(ptr)+uintptr(offset)), 0, 64)
-		}
-		C.memcpy(unsafe.Pointer(uintptr(ptr)+uintptr(offset)), dataPtr, dataSize)
-	}
-
-	return ptr, nil
-}
-
-// Append appends packets to the end of a packet table.
-func (t *Table) Append(data interface{}) (err error) {
+// Extracts the values that have to be appended to a PT. Writes them to mem, which
```

Comment:
```
// extractValues extracts the values to be appended to a packet table and adds
// them to the appendData structure.
//
// Struct values must only have exported fields, otherwise extractValues will panic.
```

Or alternatively elide those. I think the current behaviour is OK.